File: executable/executable.java
//usr/bin/env javac -d /tmp $0;java -cp /tmp `basename "$0" .java` $*;exit

import static java.lang.System.out;

public class executable {
    public static void main(String[] args) {
        if(args.length > 0) {
            out.println("arguments:");
            for(String arg : args) {
                out.println("    " + arg);
            }
        } else {
            out.println("missing arguments");
        }
    }
}

File: executable/executable.m
//usr/bin/env clang -framework Foundation -o/tmp/`basename "$0"` $0;/tmp/`basename "$0"` $*;exit

#import <Foundation/Foundation.h>

int main(int argc, char** argv) {
    if(argc > 1) {
        NSLog(@"arguments:\n");
        for(int i = 1; i < argc; ++i) {
            NSLog(@"    %s", argv[i]);
        }
        return 0;
    } else {
        NSLog(@"missing arguments");
        return 1;
    }
}
